projects/bp-gallery/src/components/views/geoguessr/GeoMap.tsx

```diff
@@ -124,20 +129,30 @@ const GeoMap = ({
 
   useEffect(() => {
     if (!map.current) return;
-    map.current.invalidateSize();
     map.current.flyTo(initialMapValues.center, initialMapValues.zoom);
   }, [pictureId, initialMapValues]);
 
+  useEffect(() => {
+    if (!map.current && !mainDivRef.current) return;
+    const resizeObserver = new ResizeObserver(() => {
+      map.current ? map.current.invalidateSize() : null;
+    });
+    // @ts-ignore
+    resizeObserver.observe(mainDivRef.current);
```

Comment:
I find the `@ts-ignore` a bit overkill for a simple `| null` problem :D
You can fix it by replacing the `&&` with a `||`. While you're at it, you can also use `map.current?.invalidateSize()` instead of the ternary operator :)